docs/database/database-metrics.md

```diff
@@ -239,6 +241,176 @@ and SHOULD be provided **at span creation time** (if provided at all):
 
 ## Experimental
 
+### Database Response
+
+The following metric instruments describe database query response.
+
+#### Metric: `db.client.response.row_count`
+
+This metric is [recommended][MetricRecommended].
+
+This metric SHOULD be specified with
+[`ExplicitBucketBoundaries`](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.35.0/specification/metrics/api.md#instrument-advisory-parameters)
+of `[1, 2, 5, 10, 20, 50, 100, 200, 500, 1000, 2000, 5000, 10000]`.
+
+Explaining bucket configuration:
+
+1. 1, 2, 5, 10, 50, 100: These buckets are useful for capturing scenarios where only a small number of items are returned. Such small queries are common in real-time or interactive applications where performance and quick responses are critical.
+2. 200, 500, 1000: These values represent typical workloads where moderate amounts of data are returned in each query.
+3. 2000, 5000: These boundaries capture scenarios where the query returns large datasets. These larger page sizes can potentially increase memory or CPU usage and may lead to longer query execution times, making it important to track performance in these ranges.
+4. 10000: This boundary is included to capture rare, very large response sizes. Such queries can put significant strain on system resources, including memory, CPU, and network bandwidth, and can often lead to performance issues such as high latency or even network drops.
+
+<!-- semconv metric.db.client.response.row_count -->
+<!-- NOTE: THIS TEXT IS AUTOGENERATED. DO NOT EDIT BY HAND. -->
+<!-- see templates/registry/markdown/snippet.md.j2 -->
+<!-- prettier-ignore-start -->
+<!-- markdownlint-capture -->
+<!-- markdownlint-disable -->
+
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `db.client.response.row_count` | Histogram | `{row}` | The actual number of records returned by the database operation. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```

Comment:
Could this be used for the number of rows affected? If not, should we make this metric name more specific to reserve a name for the rows affected? How about `db.client.response.row_returned_count`?

Reply:
`db.client.response.row_count` => it means `row_count` returned in `response` irrespective of the operations.
 Generally, row affected are equal to rows returned, if in your case, it can be different then, you can have `db.client.operation.row_count` (or in similar lines) which would indicate, how rows are affected during an operation.

Reply:
> The actual number of records returned or affected by the database operation

my understanding is this attribute is only the number of records returned in the response

and is not the number of records affected by the database operation

Reply:
@trask So, should we rename this metric to `db.client.response.row_returned_count` to make it clear? Then, the metric for records affected can be called `db.client.response.row_affected_count` in the future.

Reply:
I was thinking that the `.response.` namespace already scopes it to number of rows returned in the response, similar to `http.response.status_code` and `http.response.header.<key>`

Reply:
But we also need the response to know how many the number of affected rows are.

For instance,

```sql
UPDATE orders
SET location = "bar"
WHERE location = "foo"
```

Reply:
great catch @XSAM !

If we are able to capture both (returned and affected rows)  the value would be ambiguous.
There could be more -things to capture - e.g. we know the total number but we read only the first page.

We can have multiple metrics for them or one metric with a dimension that describes what type of row it is.
I.e.:

1. `db.client.response.returned_rows` and `db.client.response.affected_rows`, potentially `db.client.response.total_rows` 

    I think it's a viable option - we probably will have just a few metrics like this.

2. `db.client.response.row_count` with a dimension like `db.row.type|kind|?` which could have values like `affected|returned|total_returned` and may be extended further to be used for request rows too.

    With this we'll need to record multiple measurements on one metric, which is probably fine in this case. 

Overall I don't see any obvious advantages or disadvantages of one approach versus the other, but we need to pick one and either rename the metric or add a new attribute to the existing one (`db.row.type` enum with a single supported value of `returned`).

Reply:
I would go with what @lmolkova suggested, we can add a dimension, which will indicate, which kind of row numbers are these (returned vs affected). 
As of now, we can go with this metrics name and dimension, if requirement comes up, we can add the dimension

Reply:
> 2. `db.client.response.row_count` with a dimension like `db.row.type|kind|?` which could have values like `affected|returned|total_returned` and may be extended further to be used for request rows too.

I'm not sure that the aggregation across "affected" and "returned" rows is meaningful?

> 1. `db.client.response.returned_rows` and `db.client.response.affected_rows`, potentially `db.client.response.total_rows`

an advantage to this approach is that it also solves the problem of attribute names, since you might want to stamp multiple of them on a single span, e.g.

* `db.response.returned_rows`
* `db.response.affected_rows`
* `db.response.total_rows`